apps/app/src/app/pages/pages/dashboard/components/application-card/application-card.component.spec.ts

```diff
@@ -33,169 +33,213 @@ describe('ApplicationCardComponent', () => {
-    )[0];
-    expect(cardTitle.textContent).toContain(
-      application.countTasksPending?.toString()
-    );
+  describe('onClick', () => {
+    it('should emit the application', () => {
+      const application: Application = {
+        _id: {
+          applicationName: 'test',
+          applicationVersion: '1.0.0',
+        },
+      };
+      component.application = application;
+
+      const spy = spyOn(component.applicationChange, 'emit');
+      component.onClick();
+
+      expect(component.applicationChange.emit).toHaveBeenCalledWith(
+        application
+      );
+
+      spy.calls.reset();
+    });
+
+    it('should emit when button is clicked', () => {
+      const application: Application = {
+        _id: {
+          applicationName: 'test',
+          applicationVersion: '1.0.0',
+        },
+      };
+      component.application = application;
+
+      const spy = spyOn(component.applicationChange, 'emit');
+
+      const button = fixture.debugElement.nativeElement.querySelector('button');
+      button.click();
+
+      expect(component.applicationChange.emit).toHaveBeenCalledWith(
+        application
+      );
+
+      spy.calls.reset();
+    });
+  });
+
+  describe('ui', () => {
+    it('should be an article with "card" class', () => {
+      const compiled = fixture.debugElement.nativeElement;
+      expect(compiled.querySelector('article.card')).toBeTruthy();
+    });
+
+    it('should contains a h3 with "card-header" class', () => {
+      const compiled = fixture.debugElement.nativeElement;
+      expect(compiled.querySelector('h3.card-header')).toBeTruthy();
+    });
+
+    it('should contains a footer with "card-footer" class', () => {
+      const compiled = fixture.debugElement.nativeElement;
+      expect(compiled.querySelector('footer.card-footer')).toBeTruthy();
+    });
+
+    it('should contains a button element in the footer', () => {
+      const compiled = fixture.debugElement.nativeElement;
+      expect(compiled.querySelector('footer button')).toBeTruthy();
+    });
+
+    it('should have a button styled like a link', () => {
+      const compiled = fixture.debugElement.nativeElement;
+      expect(
+        compiled.querySelector('footer button').getAttribute('class')
+      ).toContain('btn btn-sm btn-link');
+    });
+
+    it('should have name and version in the header', () => {
+      component.application = {
+        _id: {
+          applicationName: 'application_1',
+          applicationVersion: '1.0.0',
+        },
+      } as Application;
+      fixture.detectChanges();
+
+      const compiled = fixture.debugElement.nativeElement;
+      expect(compiled.querySelector('h3.card-header').textContent).toContain(
+        'application_1'
+      );
+      expect(compiled.querySelector('h3.card-header').textContent).toContain(
+        '1.0.0'
+      );
+    });
+
+    it('should emit an event when the footer button is clicked', () => {
```

Comment:
Ce test là est en double du coup, tu peux l'enlever

Reply:
sur ce test là, je m'assure que le button est bien branché à la fonction 'onClick' alors que le teste ligne 37 (should emit the application) test la fonction mais du coup, le nom n'est pas le bon, je vais le changer

Reply:
Effectivement, en double, je le retire